ogre/src/OgreMaterial.cc

```diff
@@ -44,21 +44,52 @@ void OgreMaterial::Destroy()
 {
   if (!this->Scene()->IsInitialized())
     return;
-
+  std::string materialName;
   Ogre::MaterialManager &matManager = Ogre::MaterialManager::getSingleton();
 #if OGRE_VERSION_LT_1_10_1
   if (!this->ogreMaterial.isNull())
   {
+    materialName = this->ogreMaterial->getName();
+
+    this->ogreTexState->setBlank();
+    auto indexUnitStateToRemove =
+      this->ogrePass->getTextureUnitStateIndex(this->ogreTexState);
+    this->ogrePass->removeTextureUnitState(indexUnitStateToRemove);
+
     matManager.remove(this->ogreMaterial->getName());
     this->ogreMaterial.setNull();
-  }
 #else
   if (this->ogreMaterial)
   {
     matManager.remove(this->ogreMaterial->getName());
     this->ogreMaterial.reset();
   }
 #endif
+  auto &textureManager = Ogre::TextureManager::getSingleton();
+  auto iend = textureManager.getResourceIterator().end();
+  for (auto i = textureManager.getResourceIterator().begin(); i != iend;)
+  {
+      // A use count of 4 means that only RGM, RM and MeshManager have references
+      // RGM has one (this one) and RM has 2 (by name and by handle)
+      // and MeshManager keep another one int the template
+      Ogre::Resource* res = i->second.get();
+      if (i->second.useCount() == 4)
+      {
+        if (this->textureName == res->getName() &&
+          res->getName().find("scene::RenderTexture") == std::string::npos)
+        {
+          this->Scene()->ClearMaterialsCache(this->textureName);
```

Comment:
Here's an idea to prevent breaking ABI. We can add a new `OgreScene::MeshFactory()` accessor function to `OgreScene` class (and also `Ogre2Scene`) class to get the mesh factory then here we just need to do:

```
OgreScenePtr s = std::dynamic_pointer_cast<OgreScene>(this->Scene());
auto factory = s->MeshFactory();
factory->ClearMaterialCache(this->textureName);
```
This prevents adding and overriding a virtual function in Scene.